Sources/NIO/Codec.swift

```diff
@@ -89,6 +88,32 @@ public protocol ByteToMessageDecoder: ChannelInboundHandler where InboundIn == B
     func shouldReclaimBytes(buffer: ByteBuffer) -> Bool
 }
 
+extension ByteToMessageDecoder {
+    public func decoderRemoved(ctx: ChannelHandlerContext) {
+    }
+
+    public func decoderAdded(ctx: ChannelHandlerContext) {
+    }
+
+    public func shouldReclaimBytes(buffer: ByteBuffer) -> Bool {
+        return true
+    }
+
+    public func decodeLast(ctx: ChannelHandlerContext, buffer: inout ByteBuffer) throws -> DecodingState {
```

Comment:
I remain unsure about whether we should have a default implementation of this.

Reply:
we shouldn't I think. This is more mean to be the smallest diff to passing the tests. I think in decodeLast we also should make the user do something with the left-overs. Needs some design

Reply:
👍

Reply:
Deferring this for follow-up work.